pkg/backend/aws/backend.go

```diff
@@ -78,21 +74,13 @@ func NewAWSBackend(
 	}
 
 	// Create an AWS session if we don't have one
-	if b.awsSession == nil {
-		opts := session.Options{
-			Config: aws.Config{
-				Region:     b.awsRegion,
-				MaxRetries: aws.Int(2),
-			},
-			SharedConfigState: session.SharedConfigEnable,
-		}
-
-		// Set profile if we have one
-		if b.awsProfile != nil && *b.awsProfile != "" {
-			opts.Profile = *b.awsProfile
+	if b.awsConfig == nil {
+		conf, err := configv2.LoadDefaultConfig(ctx, configv2.WithRegion(*b.awsRegion), configv2.WithSharedConfigProfile(*b.awsProfile), configv2.WithRetryMaxAttempts(2))
```

Comment:
we should only set the profile if not `nil` and not `""` I think this is slightly different